nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -170,6 +178,26 @@ class SimpleActionServer
     return rclcpp_action::CancelResponse::ACCEPT;
   }
 
+  /**
+   * @brief Sets thread priority level
+   */
+  void setSoftRealTimePriority()
+  {
+    if (use_realtime_prioritization_) {
+      sched_param sch;
+      sch.sched_priority = 90;
+      if (sched_setscheduler(0, SCHED_FIFO, &sch) == -1) {
```

Comment:
This PR looks great to me. 

The only comment I have is: that `sched_setscheduler()` works only on Linux machines.
If you want support for Windows and Mac, see e.g. the function `configure_native_thread`, which @ralph-lange has written to assign the thread priority to support Linux/Windows/Mac operating system:
 
https://github.com/ros2/examples/blob/rolling/rclcpp/executors/cbg_executor/src/examples_rclcpp_cbg_executor/utilities.hpp

It is used in the callback group-level executor example:
https://github.com/ros2/examples/blob/1d97c4fc7445554f6f85f63305d424fc017212a0/rclcpp/executors/cbg_executor/src/pong.cpp#L78

You might consider using a similar utility function here as well.

Reply:
In my opinion, `configure_native_thread` should factored out for this purpose (to the rcutils package or to rclcpp?) and generalized with numeric priorities instead of the enum supporting low and high only. Happy to review such PR.

Reply:
Thanks for the remark - we don't typically think about windows/apple but if there's a utility that would allow us to have support for it easily, more than happy to ensure cross-compatibility! 

However, it looks like the signature of those tools requires an `std::thread` input, where we're using `std::async` to have a future result. It could be refactored to use `std::thread`, but it would be alot less clean.

So, even if that utility was in rclutils, it wouldn't, without modifications, be able to help us here. There would need to be modifications to the apple/qnx/windows implementations to use the current thread instead of passing in the thread object. Would that be something possible / Bosch interested in doing?